base/strings/utf8proc.jl

```diff
@@ -194,28 +200,30 @@ eltype{S}(::Type{GraphemeIterator{S}}) = SubString{S}
 function length(g::GraphemeIterator)
     c0 = Char(0x00ad) # soft hyphen (grapheme break always allowed after this)
     n = 0
+    state = Ref{Int32}(0)
     for c in g.s
-        n += isgraphemebreak(c0, c)
+        n += isgraphemebreak(c0, c, state)
         c0 = c
     end
     return n
 end
 
-start(g::GraphemeIterator) = start(g.s)
-done(g::GraphemeIterator, i) = done(g.s, i)
+start(g::GraphemeIterator) = (start(g.s), Ref{Int32}(0))
```

Comment:
Probably a good idea not to rely on the contents of iterators.  (Even for strings, the iterator may cease to be an integer someday: #9297.)